pandas/core/internals/construction.py

```diff
@@ -475,23 +473,11 @@ def dict_to_mgr(
         keys = list(data.keys())
         columns = Index(keys)
         arrays = [com.maybe_iterable_to_list(data[k]) for k in keys]
-        # GH#24096 need copy to be deep for datetime64tz case
-        # TODO: See if we can avoid these copies
         arrays = [arr if not isinstance(arr, Index) else arr._data for arr in arrays]
-        arrays = [
-            arr if not is_datetime64tz_dtype(arr) else arr.copy() for arr in arrays
-        ]
```

Comment:
Is this related to the other changes (this is not behind an `if copy` check), or just an additional copy that is no longer needed nowadays?

Reply:
this can be removed independently without breaking any tests.

Reply:
This actually causes a change in behaviour:

```
In [1]: dti = pd.date_range("2012", periods=3, tz="Europe/Brussels")

In [2]: df = pd.DataFrame({"a": dti}, copy=False)

In [3]: df.iloc[0, 0] = df.iloc[2, 0]

In [4]: df
Out[4]: 
                          a
0 2012-01-03 00:00:00+01:00
1 2012-01-02 00:00:00+01:00
2 2012-01-03 00:00:00+01:00

In [5]: dti
Out[5]: 
DatetimeIndex(['2012-01-01 00:00:00+01:00', '2012-01-02 00:00:00+01:00',
               '2012-01-03 00:00:00+01:00'],
              dtype='datetime64[ns, Europe/Brussels]', freq='D')
```

In the above, the `dti` is not mutated, but with this branch it will be. 
Now, since I explicitly passed `copy=False`, I suppose this is a bug fix :) For other Index objects it will also mutate the original index.

When the original PR (https://github.com/pandas-dev/pandas/pull/24096) added this line of code to protect DatetimeIndex from getting mutated, the `copy` behaviour of DataFrame(..) might still have been different.